pkg/tls/cred.go

```diff
@@ -83,13 +84,21 @@ func (crt *Crt) CertPool() *x509.CertPool {
 }
 
 // Verify the validity of the provided certificate
-func (crt *Crt) Verify(roots *x509.CertPool, name string) error {
+func (crt *Crt) Verify(roots *x509.CertPool, name string, currentTime time.Time) error {
 	i := x509.NewCertPool()
 	for _, c := range crt.TrustChain {
 		i.AddCert(c)
 	}
 	vo := x509.VerifyOptions{Roots: roots, Intermediates: i, DNSName: name}
```

Comment:
you need to pass the `currentTime` into the `VerifyOptions`

Reply:
yes, wasn't yet ready for review - was just trying out something else because I ran into some rebasing errors. Anyway, its ready for review now :smile: